ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/EthHashrate.java

```diff
@@ -40,10 +40,9 @@ public String getName() {
   public JsonRpcResponse response(final JsonRpcRequest req) {
     try {
       final Optional<Long> hashesPerSecond = miningCoordinator.hashesPerSecond();
-      if (hashesPerSecond.isPresent()) {
-        return new JsonRpcSuccessResponse(req.getId(), Quantity.create(hashesPerSecond.get()));
-      }
-      return new JsonRpcErrorResponse(req.getId(), JsonRpcError.NO_HASHES_PER_SECOND);
+      return hashesPerSecond
+          .map(hashes -> new JsonRpcSuccessResponse(req.getId(), Quantity.create(hashes)))
+          .orElseGet(() -> new JsonRpcSuccessResponse(req.getId(), Quantity.create(0)));
```

Comment:
`return new JsonRpcSuccessResponse(req.getId(), Quantity.create(hashesPerSecond.orElse(0))`?

Reply:
yep -- I made it better -- you've helped more -- thanks.